journal.py

```diff
@@ -86,35 +89,45 @@ def list_view(request):
 @view_config(route_name='detail', renderer='templates/detail.jinja2')
 def detail_view(request):
     entry = Entry.one(request.matchdict['id'])
-    return {'entry': entry}
+    html_text = markdown(entry.text, output_format='html5')
 
+    def my_highlight(matchobj):
+        return highlight(matchobj.group(0), PythonLexer(), HtmlFormatter())
 
-@view_config(route_name='create', renderer='templates/create.jinja2')
-def create_view(request):
-    return {}
+    pattern = r'(?<=<code>)[\s\S]*(?=<\/code>)'
+    html_text = re.sub(pattern, my_highlight, html_text)
+    return {
+        'entry': {
+            'id': entry.id,
+            'title': entry.title,
+            'text': html_text,
+            'created': entry.created
+        }
+    }
```

Comment:
This works, but you are doing a bit more work than you need to.  Check out the 'extensions' that are available for the markdown library, and in particular the [codehilite extension](https://pythonhosted.org/Markdown/extensions/code_hilite.html).  Given that, you can actually do all the work of highlighting *inside* the call to markdown.  Like so:

```
def render_markdown(content, linenums=False, pygments_style='default'):
    ext = "codehilite(linenums={linenums}, pygments_style={pygments_style})"
    out = markdown.markdown(
        content, extensions=[ext.format(**locals()), 'fenced_code']
    )
    return out
```
Not at all intuitive, and difficult to look up in the markdown documentation.  But it works!